backend/api/src/http/endpoints/user.rs

```diff
@@ -38,80 +249,82 @@ async fn user_lookup(
         .ok_or(error::UserNotFound::UserNotFound)
 }
 
-async fn validate_register_req(req: &RegisterRequest) -> Result<(), error::Register> {
-    // todo: decide if we should check for an _empty_ email?
+fn validate_register_req(req: &PutProfileRequest) -> Result<(), error::Register> {
     if req.username.is_empty() {
-        return Err(error::Register::RegisterError(
-            RegisterErrorKind::EmptyDisplayName,
-        ));
+        return Err(error::Register::EmptyUsername);
     }
 
     Ok(())
 }
 
-/// Register a new user.
-#[allow(unreachable_code, unused_variables)]
+/// Create or replace your profile.
 #[api_v2_operation]
-async fn handle_register(
+async fn put_profile(
     settings: Data<RuntimeSettings>,
     db: Data<PgPool>,
-    // todo: Option ignores errors, don't do that.
-    signup_user: OAuthSignupToken,
-    req: Json<RegisterRequest>,
+    signup_user: TokenUserWithPurposedSession<SessionPutProfile>,
+    req: Json<PutProfileRequest>,
 ) -> actix_web::Result<HttpResponse, error::Register> {
-    let signup_user = Some(signup_user);
-    match signup_user {
-        None => return Err(anyhow::anyhow!("handle authless register request").into()),
-        Some(token) => {
-            validate_register_req(&req).await?;
+    validate_register_req(&req)?;
 
-            let mut txn = db.begin().await?;
+    let mut txn = db.begin().await?;
 
-            let id = register(&mut txn, &req, &token.0.email).await?;
+    let mut upsert_txn = txn.begin().await?;
 
-            match &token.0.provider {
-                OAuthProvider::Google { google_id } => {
-                    register_google_oauth(&mut txn, &google_id, id).await?;
-                }
-            }
+    upsert_profile(&mut upsert_txn, &req, signup_user.claims.user_id).await?;
 
-            txn.commit().await?;
+    db::session::delete(&mut upsert_txn, &signup_user.claims.token).await?;
 
-            // fixme: remove the todo, remove the `#[allow]` above.
-            let (csrf, cookie) = create_signin_token(
-                id,
-                &settings.token_secret,
-                settings.is_local(),
-                TokenSource::OAuth(token.0.provider),
-                settings.login_token_valid_duration,
-            )?;
+    upsert_txn.commit().await?;
 
-            Ok(HttpResponse::Created()
-                .cookie(cookie)
-                .json(CreateSessionSuccess { csrf }))
-        }
-    }
+    let login_ttl = settings
+        .login_token_valid_duration
+        .unwrap_or(Duration::weeks(2));
+
+    let session = db::session::create(
```

Comment:
Should this route log you in?
From one perspective, you already had to verify yourself and get to this route